templates/workloads/partials/cf/subscribe.yml

```diff
@@ -0,0 +1,226 @@
+{{- if .Subscribe}}
+SQSQueue:
+  Type: AWS::SQS::Queue
+  {{- if .Subscribe.Queue}}
+  Properties:
+    QueueName: !Sub ${AWS::StackName}-{{- if .Subscribe.Queue}}{{if .Subscribe.Queue.Name}}{{logicalIDSafe .Subscribe.Queue.Name}}{{- else}}SQSQueue{{- end}}{{- if .Subscribe.Queue.FIFO}}.fifo{{- end}}{{- end}}
+    {{- if .Subscribe.Queue.Retention}}
+    MessageRetentionPeriod: {{.Subscribe.Queue.Retention}}
+    {{- end}}
+    {{- if .Subscribe.Queue.Delay}}
+    DelaySeconds: {{.Subscribe.Queue.Delay}}
+    {{- end}}
+    {{- if .Subscribe.Queue.Timeout}}
+    VisibilityTimeout: {{.Subscribe.Queue.Timeout}}
+    {{- end}}
+    {{- if .Subscribe.Queue.KMS}}
+    KmsMasterKeyId: !Ref SQSQueueKMSKey
+    {{- else}}
+    KmsMasterKeyId: 'alias/aws/sqs'
+    {{- end}}
+    {{- if .Subscribe.Queue.FIFO}}
+    FifoQueue: true
+    {{- if .Subscribe.Queue.FIFO.HighThroughput}}
+    FifoThroughputLimit: 'perMessageGroupId'
+    DeduplicationScope: 'messageGroup'
+    {{- end}}
+    {{- end}}
+    {{- if .Subscribe.Queue.DeadLetter}}
+    RedrivePolicy:
+      {{- if .Subscribe.Queue.DeadLetter.Id}}
+      deadLetterTargetArn: !Sub 'arn:${AWS::Partition}:sqs:${AWS::Region}:${AWS::AccountId}:{{logicalIDSafe .Subscribe.Queue.DeadLetter.Id}}'
+      {{- else}}
+      deadLetterTargetArn: !GetAtt DeadLetter.Arn
+      {{- end}}
+      maxReceiveCount: {{.Subscribe.Queue.DeadLetter.Tries}}
+    {{- end}}
+  {{- end}}
+
+{{- if .Subscribe.Queue}}{{- if .Subscribe.Queue.DeadLetter}}{{- if not .Subscribe.Queue.DeadLetter.Id}}
+DeadLetter:
+  Type: AWS::SQS::Queue
+  Properties:
+    QueueName: !Join ['', [!GetAtt SQSQueue.Name, "-", "DeadLetter"]]
+
+DeadLetterPolicy:
+  Type: AWS::SQS::QueuePolicy
+  Properties:
+    Queues: [!Ref 'DeadLetter']
+    PolicyDocument:
+      Version: "2012-10-17"
+      Statement:
+        - Sid: Allow-TaskRole-Subscribe
+          Effect: Allow
+          Principal:
+            Service: sns.amazonaws.com
+          Action: 
+            - sqs:GetMessage
+            - sqs:DeleteMessage
+          Resource: !GetAtt SQSQueue.Arn
+          Condition:
+            ArnEquals:
+              aws:SourceArn: !Ref 'TaskRole'
+{{- end}}{{- end}}{{- end}}
+
+{{- if .Subscribe.Queue}}{{- if .Subscribe.Queue.KMS}}
+SQSQueueKMSKey:
```

Comment:
What do you think of getting rid of `kms: true` as a field from the queues?  
We're already always encrypting with `KmsMasterKeyId: 'alias/aws/sqs'` so I'm not sure what's the benefit here of defining another one. I could imagine users maybe wanting to use their own key but we can hold off on it and it will simplify the template quite a bit

Reply:
Removed.